etl/jobs.py

```diff
@@ -0,0 +1,34 @@
+from dagster import EnvVar, RunConfig, define_asset_job
+
+from .assets import (
+    documents_of_wikipedia_articles_with_summaries,
+    wikipedia_articles_embeddings,
+    wikipedia_articles_with_summaries,
+)
+from .models import OpenAiPipelineConfig, OpenAiSettings
+from .models.types import EnrichmentType, RecordType
+
+openai_pipeline_config = OpenAiPipelineConfig(
+    openai_settings=OpenAiSettings(
+        openai_api_key=EnvVar("OPENAI_API_KEY").get_value("")
+    ),
+    record_type=EnvVar("RECORD_TYPE").get_value(default=RecordType.WIKIPEDIA),
+    enrichment_type=EnvVar("ENRICHMENT_TYPE").get_value(default=EnrichmentType.SUMMARY),
+)
+
+
+embedding_job = define_asset_job(
+    "embedding_job",
+    selection=["*" + wikipedia_articles_embeddings.key.path[0]],
+    config=RunConfig(
```

Comment:
This doesn't seem right, passing in dependencies manually. You just need to add `resources` to your `Definitions`?

Reply:
I was using `configs` initially, and this was the pattern I saw in the Dagster docs. I am using `resources` now.

Reply:
I believe `config` is for per-asset configuration rather than shared configuration.

I personally prefer to put all of the dependencies in Definitions rather than spreading them out. (You can't do everything that way -- partitions are asset-level, for example.)